ftdetect/gofiletype.vim

```diff
@@ -25,7 +25,7 @@ function! s:gofiletype_post()
 endfunction
 
 " Note: should not use augroup in ftdetect (see :help ftdetect)
-au BufNewFile *.go setfiletype go | setlocal fileencoding=utf-8 fileformat=unix
+au BufNewFile *.go setfiletype go | if &modifiable | setlocal fileencoding=utf-8 fileformat=unix | endif
```

Comment:
As mentioned in #2096, the fileencoding and fileformat options probably shouldn't be set at all here, but this is an acceptable short term solution until I can refactor `ftdetect/gofiletype.vim`, `ftplugin/go.vim` and `plugin/go.vim`.

Would you mind adding a similar guard to the `au BufNewFile *.s` line for consistency until I can do the refactor?

Reply:
Done